src/internal/proc.js

```diff
@@ -626,6 +640,18 @@ export default function proc(
     channel.flush(cb)
   }
 
+  function runGetContextEffect(prop, cb) {
+    // TODO: if object returned -> return shallow copy?
+    // should yield getContext() be available to access context?
+    // if yes then shallow copying it is a must to avoid unexpected mutations
+    cb(taskContext[prop])
+  }
+
+  function runSetContextEffect(props, cb) {
+    Object.assign(taskContext, props)
```

Comment:
such checks are usually performed in effect creators, it would only double the same check if we put it here also, as its not possible to reach this - `runSetContextEffect` in any other way that `yield setContext(...)`